.ci/test_r_package_windows.ps1

```diff
@@ -93,6 +108,10 @@ Write-Output "Installing Rtools"
 Start-Process -FilePath Rtools.exe -NoNewWindow -Wait -ArgumentList "/VERYSILENT /DIR=$env:R_LIB_PATH/Rtools" ; Check-Output $?
 Write-Output "Done installing Rtools"
 
+Write-Output "Installing dependencies"
+$packages = "c('data.table', 'jsonlite', 'Matrix', 'processx', 'R6', 'testthat'), dependencies = c('Imports', 'Depends', 'LinkingTo')"
+Run-R-Code-Redirect-Stderr "options(install.packages.check.source = 'no'); install.packages($packages, repos = '$env:CRAN_MIRROR', type = 'binary', lib = '$env:R_LIB_PATH')" ; Check-Output $?
```

Comment:
I think it may be useful to explicitly split actually required dependencies of LightGBM and other stuff. Something like the following:
```suggestion
Write-Output "Installing dependencies and other stuff"
$dependencies = @("data.table", "jsonlite", "Matrix", "R6", "testthat")
$other_packages = @("processx")
Run-R-Code-Redirect-Stderr "options(install.packages.check.source = 'no'); install.packages(c('$(($dependencies + $other_packages) -Join "', '")'), dependencies = c('Imports', 'Depends', 'LinkingTo'), repos = '$env:CRAN_MIRROR', type = 'binary', lib = '$env:R_LIB_PATH')" ; Check-Output $?
```

Reply:
I don't think there's a meaningful difference. `processx` isn't only used for CI...remember it is officially part of the package source that is used by the solution in #3065

Reply:
I thought it may be used to specify different `dependencies =` or `type =` arguments. For example, if we decide one day to install the latest version of LightGBM's deps but any version of other helper utilities.

Reply:
Ah ok. I think that we should not do this until we actually need such a split, for some of the example use cases you've mentioned. Since we currently use the same `type` and `dependencies`, I think we should leave the install code as it is.

Reply:
OK.